moon/test_controller_round3.py

```diff
@@ -0,0 +1,84 @@
+import unittest
+from unittest.mock import MagicMock
+import numpy as np
+from random import Random
+import math
+
+from shapely.geometry import LineString
+from shapely.geometry import Point
+
+try: 
+    from matplotlib import pyplot as plt
+    has_matplotlib = True
+except ImportError:
+    has_matplotlib = False
+from moon.controller_round3 import best_fit_circle
+
+def pol2cart(rho, phi):
+    x = rho * math.cos(phi)
+    y = rho * math.sin(phi)
+    return(x, y)
+
+def cart2pol(x, y):
+    rho = np.sqrt(x**2 + y**2)
+    phi = np.arctan2(y, x)
+    return(rho, phi)
+
+def plot_test_case(circle, lines):
+    circle_center_x, circle_center_y, circle_radius = circle
+    x_l, y_l = lines
+
+
+    fig, ax = plt.subplots()
+    ax.set_aspect(1)
+    ax.add_artist(plt.Circle((circle_center_x, circle_center_y), circle_radius))
+
+    for i in range(len(lines[0])):
+        plt.plot([0,x_l[i]],[0,y_l[i]])
+    plt.axis([-50, 50, -50, 50])
+    plt.show()
+
+
+class Round3CircleTest(unittest.TestCase):
+
+        
+    def test_round3_circle(self):
+        # generate random points on 2.6rad arc with random radius between 5 and 20
+        # view is at (0,0)
+        # circle is at random point and has random radius
+        rand = Random(0)
+        for circle in range(0,100):
```

Comment:
10 circles is surely sufficient ... note, that if your count goes from zero you do not have to specify it, i.e.
`for circle in range(10):`
more over if you really do not need `circle` variable you can use underscores:
`for __ in range(10):`

Reply:
ok